### PR TITLE
refs issue #7929 add support for Postgres native uuid datatype

### PR DESCRIPTION
**Pull request by skie**
Improvements for #8033 PR. Tests covering uuids are added

### CODE REVIEW

File: lib/Cake/Test/Case/Controller/Component/SecurityComponentTest.php [hunk 1]
```diff
@@ -498,10 +498,10 @@ public function testValidatePostOnGetWithData() {
 		$fields = 'an-invalid-token';
 		$unlocked = '';
 
-		$this->Controller->request->data = [
+		$this->Controller->request->data = array(
 			'Model' => array('username' => 'nate', 'password' => 'foo', 'valid' => '0'),
 			'_Token' => compact('fields', 'unlocked')
-		];
+        );
```
Comment (markstory) [inline lib/Cake/Test/Case/Controller/Component/SecurityComponentTest.php]: Indentation looks off here.

Comment (markstory) [inline lib/Cake/Test/Case/Controller/Component/SecurityComponentTest.php]: Thank you for fixing this :blush:

File: lib/Cake/Model/Datasource/Database/Postgres.php [hunk 2]
```diff
@@ -923,6 +930,19 @@ public function buildIndex($indexes, $table = null) {
 	}
 
 /**
+ * {@inheritDoc}
+ */
+	public function value($data, $column = null, $null = true) {
+		$value = parent::value($data, $column, $null);
+		if ($column == 'uuid' && is_scalar($data)) {
```
Comment (dereuromark) [inline lib/Cake/Model/Datasource/Database/Postgres.php]: `===`

File: lib/Cake/Test/Case/Model/ModelWriteTest.php [hunk 3]
```diff
@@ -293,6 +313,27 @@ public function testSaveUuidNull() {
 	}
 
 /**
+ * Ensure that if the id key is null but present the save doesn't fail (with an
+ * x sql error: "Column id specified twice")
+ *
+ * @return void
+ */
+	public function testSaveUuidNullNative() {
+		$this->skipIf(!($this->db instanceof Postgres), 'This test is compatible with Postgres only.');
```
Comment (markstory) [inline lib/Cake/Test/Case/Model/ModelWriteTest.php]: You could put these tests in the postgres driver suite instead.

Comment (skie) [inline lib/Cake/Test/Case/Model/ModelWriteTest.php]: This model layer tests. So it require fixtures loading that done in this layer only

File: lib/Cake/Model/Datasource/Database/Postgres.php (hunk 2 above)
Comment (skie) [inline lib/Cake/Model/Datasource/Database/Postgres.php]: Sorry, i did not get what "===" means

File: lib/Cake/Test/Case/Model/ModelWriteTest.php (hunk 3 above)
Comment (markstory) [inline lib/Cake/Test/Case/Model/ModelWriteTest.php]: Ok, I only suggested that as the tests only run with postgres.

**Comment by markstory** [comment]
This was closed when the 2.8 branch was closed off. Would it be possible for you to open a new pull request against the `2.x` branch?

**Pull request closed without merging**